src/ui/components/componenttypes.js

```diff
@@ -0,0 +1,13 @@
+/** @module */
+
+// TODO add all component types
```

Comment:
Add a little JSDoc explaining what this enum is (don't need much, fairly self explanatory)